icu4c/source/i18n/collationdatabuilder.cpp

```diff
@@ -255,12 +255,17 @@ DataBuilderCollationIterator::getDataCE32(UChar32 c) const {
 
 uint32_t
 DataBuilderCollationIterator::getCE32FromBuilderData(uint32_t ce32, UErrorCode &errorCode) {
+    if (U_FAILURE(errorCode)) { return 0; }
     U_ASSERT(Collation::hasCE32Tag(ce32, Collation::BUILDER_DATA_TAG));
     if((ce32 & CollationDataBuilder::IS_BUILDER_JAMO_CE32) != 0) {
         UChar32 jamo = Collation::indexFromCE32(ce32);
         return utrie2_get32(builder.trie, jamo);
     } else {
         ConditionalCE32 *cond = builder.getConditionalCE32ForCE32(ce32);
+        if (cond == nullptr) {
+            errorCode = U_INVALID_STATE_ERROR;
```

Comment:
```suggestion
            errorCode = U_INTERNAL_PROGRAM_ERROR;
```
These changes are good for preventing a crash, but please submit a follow-up ticket for figuring out why this happens, and put a comment before this line like
```
        // TODO: ICU-xxxxx figure out why this happens.
```